### PR TITLE
feat(Title1,Title2): new components

### PR DESCRIPTION
**Pull request by atabel**
WEB-550

### CODE REVIEW

File: doc/layout.md [hunk 1]
```diff
@@ -316,14 +316,14 @@ This is how a page layout could look like:
   <Box spaceY={24}>
     <Stack space={32}>
       <Stack space={16}>
-        <SectionTitle>Section 1</SectionTitle>
+        <Title1>Section 1</Title1>
```
Comment (atabel) [inline doc/layout.md:319]: I replaced all the internal usages of SectionTitle to Title1

File: src/title.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+};
+
+type TitleProps = {
+    children: React.ReactNode;
+    id?: string;
+    right?: React.ReactNode;
+    as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
+};
+
+export const Title1 = ({children, as = 'h3', id, right}: TitleProps): React.ReactElement => {
```
Comment (atabel) [inline src/title.tsx:56]: I decided to keep the default `as="h3"` from `SectionTitle`.

File: src/title.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    const theme = useTheme();
+
+    return (
+        <TitleLayout
+            desktopLinkHeight={20}
+            mobileLinkHeight={16}
```
Comment (atabel) [inline src/title.tsx:62]: these magic values are the line-height of `Text1` in desktop/mobile, I didn't find an automatic way of doing this without having to repeat these values here.

File: src/title.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            right={right}
+        />
+    );
+};
+
+export const Title2 = ({children, as = 'h3', id, right}: TitleProps): React.ReactElement => {
+    return (
+        <TitleLayout
+            desktopLinkHeight={32}
+            mobileLinkHeight={24}
```
Comment (atabel) [inline src/title.tsx:77]: same here but for `Text5` `line-height`

File: src/title.tsx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+import {Text1, Text2, Text5} from './text';
+import {useTheme} from './hooks';
+import Inline from './inline';
+import {createUseStyles} from './jss';
+import {pxToRem} from './utils/css';
+
+const useStyles = createUseStyles((theme) => ({
+    right: {
+        display: 'flex',
+        alignItems: 'center',
```
[the remark this answers is not in the document]
Comment (atabel) [reply src/title.tsx:11]: `SectionTitle` was centered

[the remark this answers is not in the document]
Comment (atabel) [reply src/title.tsx:11]: ok, I'll give it a try

**Review by AnaMontes11 — APPROVED** [review]
It looks good!

File: src/title.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (yceballost) [reply src/title.tsx]: yes, in the section title this "extra" space is distributed at the top and the bottom of the title and now all the extra space is on the top of the title, right? but we can live with that IMO. Do you anticipate problems?

File: src/__stories__/title-story.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            right={linkText ? <TextLink href="https://google.com">{linkText}</TextLink> : undefined}
+        >
+            {title}
+        </Title2>
+    </Stack>
+);
+
+Default.storyName = 'Title';
+Default.args = {
+    title: 'This is a title',
```
Comment (pladaria) [inline src/__stories__/title-story.tsx:29]: I'd set the default arg value for title to empty string and inside each example:
```
{title || 'Title1'}
{title || 'Title2'}
```
This way in the story you know what component are you seeing

**Review by yceballost — APPROVED** [review]
thanks!

File: src/__stories__/title-story.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+type Args = {title: string; linkText: string};
+
+export const Default: StoryComponent<Args> = ({title, linkText}) => (
+    <Stack space={32}>
+        <Title1
+            as="h3"
+            right={linkText ? <TextLink href="https://google.com">{linkText}</TextLink> : undefined}
+        >
+            {title || 'Title 1'}
```
Comment (pladaria) [inline src/__stories__/title-story.tsx:16]: Title1, without space

**Review by RubenMorPov — APPROVED** [review]

**Review by pladaria — APPROVED** [review]

**Pull request merged**